packages/react/src/primitives/TextArea/AutoresizeTextarea.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { ForwardRefPrimitive, Primitive } from '../types/view';
+import { BaseTextAreaProps, TextAreaProps } from '../types/textArea';
+import { primitiveWithForwardRef } from '../utils/primitiveWithForwardRef';
+import { TextArea } from './TextArea';
+import { useAutoresizeTextArea } from './useAutoresizeTextarea';
+import { useComposeRefsCallback } from '../../hooks/useComposeRefsCallback';
+
+const AutoresizeTextAreaPrimitive: Primitive<TextAreaProps, 'textarea'> = (
+  { value, ...rest },
+  externalRef
+) => {
+  const internalRef = React.useRef<HTMLTextAreaElement>(null);
+  useAutoresizeTextArea(internalRef.current, value as string);
```

Comment:
What happens if `value` is not a `string` (e.g. `undefined`)?

Reply:
Good point, removing the casting and really the type doesn't matter we only need value to include in the dependency array of an effect

Reply:
done